### ISSUE TITLE
[dev] cannot inherit dist path from 'build.publish'

### ISSUE THREAD

**Issue by kripod** [netlify/cli#624]
**Do you want to request a _feature_ or report a _bug_?**

I would like to report a bug.

**What is the current behavior?**

When executing `netlify dev` without a given `dev.publish` setting, e.g. with the `netlify.toml` configuration below:

```toml
[build]
  command = 'yarn build'
  functions = './server/dist/functions/'
  publish = './client/public/'

[build.environment]
  NODE_ENV = 'production'

[dev]
  command = 'wsrun --no-prefix develop'
  port = 8000
  targetPort = 8000
  functionsPort = 4000
```

It results in the following error, although the `build.publish` setting [should be inherited](https://github.com/netlify/cli/blob/02a8f3232b040a7dcc6da83251deff94960c7615/src/commands/dev/index.js#L343) by the `dev` command:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at module.exports (D:/Development/Projects/exigo/node_modules/netlify-cli/src/utils/rules-proxy.js:82:12)
    at startProxy (D:/Development/Projects/exigo/node_modules/netlify-cli/src/commands/dev/index.js:136:20)
    at async DevCommand.run (D:/Development/Projects/exigo/node_modules/netlify-cli/src/commands/dev/index.js:407:25)
    at async DevCommand._run (D:/Development/Projects/exigo/node_modules/@oclif/command/lib/command.js:44:20)
```

The issue can be reproduced through the [public repository of the exigo project](https://github.com/kripod/exigo).

**What is the expected behavior?**

The `dist` setting should be assigned automatically from the `build` configuration right [here](https://github.com/netlify/cli/blob/42f2187d96d44a8f7fd4181a047f086dc01ae5e4/src/utils/detect-server.js#L89).

**Local Environment Information**

```
  System:
    OS: Windows 10 10.0.17134
    CPU: (4) x64 Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz
  Binaries:
    Node: 12.11.1 - C:\Users\kripod\AppData\Local\Temp\yarn--1574981709087-0.4250520225519725\node.CMD
    Yarn: 1.19.1 - C:\Users\kripod\AppData\Local\Temp\yarn--1574981709087-0.4250520225519725\yarn.CMD
    npm: 6.11.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 42.17134.1038.0
```

**Comment by RaeesBhatti** [comment]
@kripod Thank you for reporting this issue. Can you please update to `netlify-cli@2.45.0` and try again. See if that fixes it.

**Comment by RaeesBhatti** [comment]
The `dist` directory comes from the framework configuration detector. It seems like in your case, it's defaulting to static server which runs in current directory.

**Comment by tibotiber** [comment]
fwiw, I had the same issue (also using `wsrun`) and updating fixed the issue. I'm guessing this is linked to #809.

**Comment by RaeesBhatti** [comment]
This has been fixed in latest version `netlify-cli@2.50.0`. Please let us know if you're still facing this problem!
